### PR TITLE
Add RPM tests for SP3, SP4, SP5

### PR DESCRIPTION
**Pull request by rtorrero**
This PR adds some basic testing for the playbooks, using the RPM installation method in
 - SP3
 - SP4
 - SP5
 
 It: 
  - Runs `playbook.yml`
  - Does a simple curl test to the readyness endpoint (`/api/readyz`
  - Runs `playbook.cleanup.yml`
  
  3 new machines have been deployed in azure for this purpose. I'm currently adding the required variables to the repository secrets

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -62,6 +62,273 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
 
+  deploy-test-sp3-rpm:
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:65]: Doesn't look this as a perfect place to use a `matrix`?
I mean, the unique difference is the IP address as far as I see

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -62,6 +62,273 @@ jobs:
...
+    name: Run the playbook on SLES 15 SP3
+    runs-on: ubuntu-latest
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:67]: Let's avoid using `latest`. Let's pin to some existing tag

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -62,6 +62,273 @@ jobs:
...
+        run: curl -k "https://$TEST_SP4_HOST_IP/api/readyz"
+      - name: Run playbook cleanup
+        uses: dawidd6/action-ansible-playbook@v2
+        with:
+          playbook: playbook.cleanup.yml
+          key: ${{ secrets.SSH_MACHINE_KEY }}
+          inventory: |
+            all:
+              vars:
+                ansible_user: ec2-user
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:212]: this `ec2-user` should be a variable.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -62,6 +62,273 @@ jobs:
...
+              children:
+                trento-server:
+                  hosts:
+                    server:
+                      ansible_host: ${{ secrets.TEST_SP4_HOST_IP }}
+                      ansible_user: ${{ secrets.TEST_HOST_USER }}
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:218]: Do we need this `ansible_user` here?
You don't use it anywhere else

**Review by arbulu89 — CHANGES_REQUESTED** [review]
The check looks good!
But let's use a matrix, it looks the perfect place for that

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (rtorrero) [reply .github/workflows/ci.yaml]: Also a leftover, fixed

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (rtorrero) [reply .github/workflows/ci.yaml]: This was a leftover, fixed

**Review by arbulu89 — APPROVED** [review]
Great! Thank you!
Merge if the stage is green!

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -62,12 +62,22 @@ jobs:
...
-  deploy-test-sp3-rpm:
-    name: Run the playbook on SLES 15 SP3
-    runs-on: ubuntu-latest
+  deploy-test-rpm:
+    name: Run the playbook on SLES 15 ${{ matrix.sp_version }}
+    runs-on: ubuntu-22.04
     needs: [build-and-push-container-images]
+    strategy:
+      matrix:
+        sp_version: ['SP3', 'SP4', 'SP5']
```
[the remark this answers is not in the document]
Comment (rtorrero) [reply .github/workflows/ci.yaml:71]: mmmh, I was trying to do this -> https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-expanding-configurations

but I guess you can't reference secrets inside a matrix ?

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -62,6 +62,104 @@ jobs:
...
+        include:
+          - sp_version: 'SP3'
+            host_ip: ${{ secrets.TEST_SP3_HOST_IP }}
+          - sp_version: 'SP4'
+            host_ip: ${{ secrets.TEST_SP4_HOST_IP }}
+          - sp_version: 'SP5'
+            host_ip: ${{ secrets.TEST_SP5_HOST_IP }}
+    env:
+      TEST_HOST_IP: ${{ matrix.host_ip }}
```
[the remark this answers is not in the document]
Comment (rtorrero) [reply .github/workflows/ci.yaml:80]: Normally we should, but one of the uses is in a `curl` call, and there it needs to be inside an env var (can't reference secrets from bash)

File: .github/workflows/ci.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rtorrero) [reply .github/workflows/ci.yaml]: Ah yeah, it's not needed in this case :+1:

**Review by EMaksy — APPROVED** [review]
LGTM!

**Pull request merged**